source/math/FloatRotation3D.ooc

```diff
@@ -0,0 +1,58 @@
+//
+// Copyright (c) 2011-2014 Simon Mika <simon@mika.se>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+import math
+import FloatExtension
+import FloatPoint3D
+import FloatTransform2D
+
+FloatRotation3D: cover {
+	x, y, z: Float
+	init: func@ ~full(=x, =y, =z)
+	init: func@ ~default { this init(0.0f, 0.0f, 0.0f) }
+	init: func@ ~fromPoint(point: FloatPoint3D) { this init(point x, point y, point z) }
+	getTransform: func(k: Float) -> FloatTransform2D {
+			rx := FloatTransform2D createXRotation(this x, k)
+			ry := FloatTransform2D createYRotation(this y, k)
+			rz := FloatTransform2D createZRotation(this z)
+			return rz * rx * ry // Yaw -> Pitch -> Roll
+	}
+	round: func -> This { This new(this x round(), this y round(), this z round()) }
+	ceiling: func -> This { This new(this x ceil(), this y ceil(), this z ceil()) }
+	floor: func -> This { This new(this x floor(), this y floor(), this z floor()) }
+	minimum: func (ceiling: This) -> This { This new(this x minimum(ceiling x), this y minimum(ceiling y), this z minimum(ceiling z)) }
+	maximum: func (floor: This) -> This { This new(this x maximum(floor x), this y maximum(floor y), this z maximum(floor z)) }
+	clamp: func ~point(floor, ceiling: This) -> This { This new(this x clamp(floor x, ceiling x), this y clamp(floor y, ceiling y), this z clamp(floor z, ceiling z)) }
+	clamp: func ~float(floor, ceiling: Float) -> This { This new(this x clamp(floor, ceiling), this y clamp(floor, ceiling), this z clamp(floor, ceiling)) }
+	operator + (other: This) -> This { This new(this x + other x, this y + other y, this z + other z) }
+	operator - (other: This) -> This { This new(this x - other x, this y - other y, this z - other z) }
+	operator - -> This { This new(-this x, -this y, -this z) }
+	operator * (other: This) -> This { This new(this x * other x, this y * other y, this z * other z) }
+	operator / (other: This) -> This { This new(this x / other x, this y / other y, this z / other z) }
+	operator * (other: Float) -> This { This new(this x * other, this y * other, this z * other) }
+	operator / (other: Float) -> This { This new(this x / other, this y / other, this z / other) }
```

Comment:
Unless you know what you're doing with this and didn't just copy it from `FloatPoint3D` or something, I would recommend removing all of these arithmetic functions. What does it even mean to multiply one `FloatRotation3D` with another?